### PR TITLE
Feature/nicer loaders

### PR DESCRIPTION
**Pull request by martynaspuskunigis**

### CODE REVIEW

File: src/app/components/recipe/recipes-item-view.tsx [hunk 1]
```diff
@@ -30,44 +32,64 @@ export class RecipesItemView extends React.Component<Props> {
...
-                                "far fa-star": !this.props.isFavorite,
-                                "fas fa-star": this.props.isFavorite
-                            },
-                            {
-                                "star-icon-empty": !this.props.isFavorite,
-                                "star-icon-full": this.props.isFavorite
-                            }
-                        )}
-                        onClick={event => this.handleFavoriteClick(event, this.props.recipe.recipe_id)}
+        if (this.props.recipe != null && this.props.isFavorite != null) {
```
Comment (DonatasBrazionis) [inline src/app/components/recipe/recipes-item-view.tsx]: Maybe use fail-fast principle? 
https://dzone.com/articles/fail-fast-principle-in-software-development

File: src/app/components/recipe/recipes-item-view.tsx [hunk 2]
```diff
@@ -30,44 +32,64 @@ export class RecipesItemView extends React.Component<Props> {
...
+                            {this.props.recipe.title}
+                        </a>
+                        <ReactTooltip id={`${this.props.recipe.recipe_id}`} type="dark" effect="float">
+                            <span>{this.props.recipe.title}</span>
+                        </ReactTooltip>
+                        <p className="recipes-subtitle">
+                            Publisher: <span>{this.props.recipe.publisher}</span>
+                        </p>
+                        <div
+                            className={classNames(
```
Comment (DonatasBrazionis) [inline src/app/components/recipe/recipes-item-view.tsx]: Maybe use `className={ this.props.isFavorite ? "fas fa-star star-icon-full" : "far fa-star star-icon-empty"}`?

File: src/app/components/recipe/recipes-item-view.tsx [hunk 3]
```diff
@@ -30,44 +32,64 @@ export class RecipesItemView extends React.Component<Props> {
...
+                                    "far fa-star": !this.props.isFavorite,
+                                    "fas fa-star": this.props.isFavorite
+                                },
+                                {
+                                    "star-icon-empty": !this.props.isFavorite,
+                                    "star-icon-full": this.props.isFavorite
+                                }
+                            )}
+                            onClick={event => this.handleFavoriteClick(event, this.props.recipe!.recipe_id)}
+                            //Exclamation mark because I am checking in the beginning whether this.props.recipe is undefined or null
```
Comment (DonatasBrazionis) [inline src/app/components/recipe/recipes-item-view.tsx]: Comment in wrong place. 
End comment with dot `.`.
Missing empty space in comment start.

File: src/app/components/recipe/recipes-item-view.tsx (hunk 2 above)
Comment (martynaspuskunigis) [reply src/app/components/recipe/recipes-item-view.tsx]: Fixed in another pull request.

**Pull request merged**